rbac/common/key/add_key.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2018 Contributors to Hyperledger Sawtooth
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# -----------------------------------------------------------------------------
+""" Registers a new public key for a user
+    usage: rbac.key.add()
+"""
+import logging
+
+from rbac.common import addresser
+from rbac.common.base.base_message import BaseMessage
+
+LOGGER = logging.getLogger(__name__)
+
+
+class AddKey(BaseMessage):
+    """ Implements the ADD_KEY message
+    """
+
+    def __init__(self):
+        super().__init__()
+        self._register()
+
+    @property
+    def message_action_type(self):
```

Comment:
would it make more sense to pass these types into the __init__ method? It seems like a lot of ceremony for each message.

Reply:
Yes, that could work as well. Explicit properties let's us mark them as being required to be implemented in the abstract/base class, and we have intermediate base classes, so this pattern seemed to work better.